server/src/main/java/app/user/UserRepository.java

```diff
@@ -1,9 +1,17 @@
 package app.user;
 
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+import org.springframework.data.domain.Pageable;
 
 import java.util.Optional;
+import java.util.List;
 
 public interface UserRepository extends JpaRepository<User, Integer> {
     Optional<User> findByEmail(String email);
+
+    @Query("SELECT u FROM User u WHERE LOWER(u.name) LIKE LOWER(CONCAT('%', :query, '%')) ORDER BY u.name ASC")
+    List<User> findByNameContainingIgnoreCase(@Param("query") String query, Pageable pageable);
```

Comment:
instead try to

public interface UserRepository extends JpaRepository<User, Long> {
    List<User> findByNameContainingIgnoreCase(String name, Sort sort);
}

@Autowired
private UserRepository userRepository;

public List<User> searchUsersSorted(String keyword) {
    Sort sort = Sort.by(Sort.Direction.ASC, "name"); // Sort by name in ascending order
    return userRepository.findByNameContaining(keyword, sort);
}


sources : 

https://chatgpt.com/share/675574d5-fae8-800d-b3b3-90ea62ff1739

I have never used Jpa with sql before so everything should be googled and double checked